L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -0,0 +1,1100 @@
+    {
+        std::string line(&buffer[0]);
+        // Extract keys consisting of the module id
+        // and of the trigger cell id
+        // Match patterns (X,Y) 
+        // where X is a number with less than 4 digis
+        // and Y is a number with less than 4 digits
+        std::regex key_regex("\\(\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*\\)");
+        std::vector<std::string> key_tokens {
+            std::sregex_token_iterator(line.begin(), line.end(), key_regex), {}
+        };
+        if(key_tokens.empty())
+        {
+            throw cms::Exception("BadGeometry")
+                << "Syntax error in the L1TCellNeighborsMapping:\n"
+                << "  Cannot find the trigger cell key in line:\n"
+                << "  '"<<&buffer[0]<<"'\n";
+        }
+        std::regex digits_regex("\\d{1,3}");
+        std::vector<std::string>  module_tc {
+            std::sregex_token_iterator(key_tokens[0].begin(), key_tokens[0].end(), digits_regex), {}
+        };
+        // get module and cell id 
+        int module = std::stoi(module_tc[0]);
+        int trigger_cell = std::stoi(module_tc[1]);
+        unsigned map_key = packTriggerCell(module, trigger_cell);
+        // Extract neighbors
+        // Match patterns (X,Y) 
+        // where X is a number with less than 4 digits
+        // and Y is a number with less than 4 digits
+        std::regex neighbors_regex("\\(\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*\\)");
+        std::vector<std::string> neighbors_tokens {
+            std::sregex_token_iterator(line.begin(), line.end(), neighbors_regex), {}
+        };
+        if(neighbors_tokens.size()<2)
+        {
+            throw cms::Exception("BadGeometry")
+                << "Syntax error in the L1TCellNeighborsMapping:\n"
+                << "  Cannot find any neighbor in line:\n"
+                << "  '"<<&buffer[0]<<"'\n";
+        }
+        auto itr_insert = neighbors_map.emplace(map_key, std::set<std::pair<short,short>>());
+        // The first element is the key, so start at index 1
+        for(unsigned i=1; i<neighbors_tokens.size(); i++)
+        {
+            const auto& neighbor = neighbors_tokens[i];
+            std::vector<std::string>  pair_neighbor {
+                std::sregex_token_iterator(neighbor.begin(), neighbor.end(), digits_regex), {}
+            };
+            short neighbor_module(std::stoi(pair_neighbor[0]));
+            short neighbor_cell(std::stoi(pair_neighbor[1]));
+            itr_insert.first->second.emplace(neighbor_module, neighbor_cell);
+        }
+    }
+    if(!l1tCellNeighborsMappingStream.eof()) edm::LogWarning("HGCalTriggerGeometry") << "Error reading L1TCellNeighborsMapping'\n";
+    l1tCellNeighborsMappingStream.close();
+
+}
+
+
+void 
+HGCalTriggerGeometryV9Imp1::
+fillNeighborMapScintillator(const edm::FileInPath& file,  std::unordered_map<int, std::set<std::pair<short,short>>>& neighbors_map)
```

Comment:
this function is nearly identical to the previous one (just a few lines differ); they should be combined